flake8_trio.py

```diff
@@ -1421,9 +1448,89 @@ def visit_AsyncFunctionDef(
 
     def visit_Call(self, node: ast.Call):
         if self.async_function and not getattr(node, "awaited", False):
-            blocking_calls = self.options.trio200_blocking_calls
-            if key := fnmatch_qualified_name([node.func], *blocking_calls):
-                self.error(node, key, blocking_calls[key])
+            self.visit_blocking_call(node)
+
+    def visit_blocking_call(self, node: ast.Call):
+        blocking_calls = self.options.trio200_blocking_calls
+        if key := fnmatch_qualified_name([node.func], *blocking_calls):
+            self.error(node, key, blocking_calls[key])
+
+
+@error_class
+class Visitor21X(Visitor200):
+    error_codes = {
+        "TRIO210": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO211": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO212": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+    }
+
+    def visit_blocking_call(self, node: ast.Call):
+        http_methods = (
+            "get",
+            "options",
+            "head",
+            "post",
+            "put",
+            "patch",
+            "delete",
+            "request",
+            "send",
+        )
+        for http_package in "requests", "httpx":
+            if get_matching_call(node, *http_methods, base=http_package):
+                self.error(node, ast.unparse(node.func), error_code="TRIO210")
+                return
+
+        func_name = ast.unparse(node.func)
+        if func_name == "urllib3.requests":
```

Comment:
so simply match on
```python
isinstance(arg, ast.Attribute) and arg.attr == 'request'
```
and error? Or how do you want to care for "GET"/"POST"

Reply:
That plus "the first argument is a string literal which is in GET/POST/PUT/etc".  This could still be a false alarm, or miss e.g. `urllib3.request(method_variable, ...)`, so we should keep the existing checks as-is and put this under a new error code.

Reply:
What about
```python
client = httpx.Client()
client.get('https://example.org')
```

Reply:
> That plus "the first argument is a string literal which is in GET/POST/PUT/etc". This could still be a false alarm, or miss e.g. `urllib3.request(method_variable, ...)`, so we should keep the existing checks as-is and put this under a new error code.

oh and we can detect urllib3 import to make this less noisy, same with httpx.

Reply:
> What about
> 
> ```python
> client = httpx.Client()
> client.get('https://example.org')
> ```

Seems tricky to implement, so let's rule it out-of-scope for this PR, and we can reconsider once we get the rest of #58  in.